test/core/crawlers/http_crawler.test.ts

```diff
@@ -0,0 +1,153 @@
+import { HttpCrawler } from '@crawlee/http';
+import type { AddressInfo } from 'net';
+import http from 'http';
+import { MemoryStorageEmulator } from 'test/shared/MemoryStorageEmulator';
+
+const router = new Map<string, http.RequestListener>();
+router.set('/', (req, res) => {
+    res.setHeader('content-type', 'text/html; charset=utf-8');
+    res.end(`<html><head><title>Example Domain</title></head></html>`);
+});
+
+router.set('/hello.html', (req, res) => {
+    res.end(`<html><head><title>Example Domain</title></head></html>`);
+});
+
+router.set('/noext', (req, res) => {
+    res.end(`<html><head><title>Example Domain</title></head></html>`);
+});
+
+router.set('/invalidContentType', (req, res) => {
+    res.setHeader('content-type', 'crazy-stuff; charset=utf-8');
+    res.end(`<html><head><title>Example Domain</title></head></html>`);
+});
+
+let server: http.Server;
+let url: string;
+
+beforeAll((cb) => {
+    server = http.createServer((request, response) => {
+        try {
+            const requestUrl = new URL(request.url, 'http://localhost');
+            router.get(requestUrl.pathname)(request, response);
+        } catch (error) {
+            response.destroy();
+        }
+    });
+
+    server.listen(() => {
+        url = `http://127.0.0.1:${(server.address() as AddressInfo).port}`;
+        cb();
+    });
+});
+
+afterAll((cb) => {
+    server.close(cb);
+});
+
+const localStorageEmulator = new MemoryStorageEmulator();
+
+beforeEach(async () => {
+    await localStorageEmulator.init();
+});
+
+afterEach(async () => {
+    await localStorageEmulator.destroy();
+});
+
+test('works', async () => {
+    const results: string[] = [];
+
+    const crawler = new HttpCrawler({
+        maxRequestRetries: 0,
+        requestHandler: ({ body }) => {
+            results.push(body as string);
+        },
+    });
+
+    await crawler.run([url]);
+
+    expect(results[0].includes('Example Domain')).toBeTruthy();
+});
+
+test('should parse content type from header', async () => {
+    const results: {type: string; encoding: BufferEncoding}[] = [];
```

Comment:
nit: this is against the code style we use. like if we dont lint this? weird
```suggestion
    const results: { type: string; encoding: BufferEncoding }[] = [];
```